fast64_internal/sm64/sm64_objects.py

```diff
@@ -1403,6 +1421,719 @@ def draw_act(self, obj, layout, value):
+                    break
+            export_line = match_line - 1
+        elif sig_insert_line:
+            export_line = sig_insert_line
+        else:
+            export_line = default_line + 3 if default_line else len(bhv_data_lines)
+            bhv_data_lines.insert(export_line, f"\n{fast64_sig}\n")
+        bhv_data_lines.insert(export_line + 1, export_bhv_name)
+
+        for j, bhv_cmd in enumerate(props.behavior_script):
+            if bhv_cmd.inherit_from_export:
+                args = bhv_cmd.get_inherit_args(context, props)
+            elif bhv_cmd.macro_args:
+                args = ", ".join(bhv_cmd.macro_args)
+            else:
+                args = ""
+            bhv_macro = f"    {bhv_cmd.macro}({args}),\n"
+            bhv_data_lines.insert(export_line + 2 + j, bhv_macro)
+        bhv_data_lines.insert(export_line + 3 + j, "};\n\n")
+        self.write_file_lines(behavior_data, bhv_data_lines)
+
+    # verify you can run this operator
+    def verify_context(self, context, props):
+        if context.mode != "OBJECT":
+            raise PluginError("Operator can only be used in object mode.")
+        if context.scene.fast64.sm64.exportType != "C":
+            raise PluginError("Combined Object Export only supports C exporting")
+        if not props.col_object and not props.gfx_object:
+            raise PluginError("No export object selected")
+
+    def get_export_objects(self, context, props):
+        if not props.export_all_selected:
+            return {props.col_object, props.gfx_object}
+
+        def obj_root(object, context):
+            while object.parent and object.parent in context.selected_objects:
+                if object.parent_type in {"ARMATURE", "OBJECT"}:
+                    return obj_root(object.parent, context)
+                else:
+                    return object
+            return object
+
+        root_objects = {obj_root(obj, context) for obj in context.selected_objects}
+        actor_objs = []
+        for obj in root_objects:
+            # eval this
+            if "Geo" in obj.sm64_obj_type or obj.sm64_obj_type in {"None", "Switch"}:
+                actor_objs.append(obj)
+
+        return actor_objs
+
+    # writes collision.inc.c file, collision_header.h
+    # writes include into aggregate file in export location (leveldata.c/<group>.c)
+    # writes name to header in aggregate file location (actor/level)
+    # var name is: const Collision <props.col_obj>_collision[]
+    def execute_col(self, props, context, obj):
+        try:
+            if props.export_col and props.obj_name_col:
+                bpy.ops.object.sm64_export_collision(export_obj=obj.name)
+        except Exception as e:
+            # pass on multiple export, throw on singular
+            if props.export_all_selected:
+                return
+            else:
+                raise Exception(e)
```

Comment:
Why not just do if not props.export_all_selected

Reply:
Unresolving this, but also I personally do this in these kinds of exceptions as much as possible, in here it may seem less useful but having a full traceback is usually good, pylint warns for it
```suggestion
        except Exception as exc:
            # pass on multiple export, throw on singular
            if not props.export_all_selected:
                raise Exception(exc) from exc
```